File: Growl Connectors/NET/Sample App/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Growl.Connector;

namespace GrowlNETConnector.Sample
{
    public partial class Form1 : Form
    {
        private GrowlConnector growl;
        private NotificationType notificationType;
        private Growl.Connector.Application application;
        private string sampleNotificationType = "SAMPLE_NOTIFICATION";

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            this.notificationType = new NotificationType(sampleNotificationType, "Sample Notification");

            this.growl = new GrowlConnector();
            //this.growl = new GrowlConnector("password");    // use this if you need to set a password - you can also pass null or an empty string to this constructor to use no password
            //this.growl = new GrowlConnector("password", "hostname", GrowlConnector.TCP_PORT);   // use this if you want to connect to a remote Growl instance on another machine

            this.growl.NotificationCallback += new GrowlConnector.CallbackEventHandler(growl_NotificationCallback);

            // set this so messages are sent in plain text (easier for debugging)
            this.growl.EncryptionAlgorithm = Cryptography.SymmetricAlgorithmType.PlainText;
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.application = new Growl.Connector.Application(this.textBox1.Text);

            this.growl.Register(application, new NotificationType[] { notificationType });
        }

        private void button2_Click(object sender, EventArgs e)
        {
            CallbackContext callbackContext = new CallbackContext();
            callbackContext.Data = "some fake information";
            callbackContext.Type = "fake data";

            Notification notification = new Notification(this.application.Name, this.notificationType.Name, DateTime.Now.Ticks.ToString(), this.textBox2.Text, this.textBox3.Text);
            this.growl.Notify(notification, callbackContext);
        }

        void growl_NotificationCallback(Response response, CallbackData callbackData)
        {
            string text = String.Format("Response Type: {0}\r\nNotification ID: {1}\r\nCallback Data: {2}\r\nCallback Data Type: {3}\r\n", callbackData.Result, callbackData.NotificationID, callbackData.Data, callbackData.Type);
            MessageBox.Show(text, "Callback received", MessageBoxButtons.OK, MessageBoxIcon.Information, MessageBoxDefaultButton.Button1);
        }
    }
}

File: Growl Connectors/NET/VB.NET Example/Form1.vb
Imports Growl.Connector

Public Class Form1

    Dim WithEvents growl As GrowlConnector
    Dim nt As NotificationType
    Dim app As Growl.Connector.Application
    Dim sampleNotificationType As String = "SAMPLE_NOTIFICATION"


    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.nt = New NotificationType(Me.sampleNotificationType, "Sample Notification")

        Me.growl = New GrowlConnector()
        'Me.growl = New GrowlConnector("password")    ' use this if you need to set a password - you can also pass null or an empty string to this constructor to use no password
        'Me.growl = New GrowlConnector("password", "hostname", GrowlConnector.TCP_PORT)   ' use this if you want to connect to a remote Growl instance on another machine

        ' set this so messages are sent in plain text (easier for debugging)
        Me.growl.EncryptionAlgorithm = Cryptography.SymmetricAlgorithmType.PlainText
    End Sub


    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        Me.app = New Growl.Connector.Application(Me.textBox1.Text)
        Dim types() As NotificationType = New NotificationType() {Me.nt}

        Me.growl.Register(Me.app, types)

        Me.Panel1.Visible = True
    End Sub

    Private Sub button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button2.Click
        Dim cc As CallbackContext = New CallbackContext("some fake information", "fake data")

        Dim n As New Notification(Me.app.Name, Me.nt.Name, DateTime.Now.Ticks.ToString(), Me.textBox2.Text, Me.textBox3.Text)
        Me.growl.Notify(n, cc)
    End Sub

    Private Sub growl_NotificationCallback(ByVal response As Response, ByVal callbackData As CallbackData, ByVal state As Object) Handles growl.NotificationCallback
        Dim text As String = String.Format("Response Type: {0} - Notification ID: {1} - Callback Data: {2} - Callback Data Type: {3}", callbackData.Result, callbackData.NotificationID, callbackData.Data, callbackData.Type)
        MessageBox.Show(text, "Callback received", MessageBoxButtons.OK, MessageBoxIcon.Information, MessageBoxDefaultButton.Button1)
    End Sub

End Class
